test/integration/node-specific/mongo_client.test.ts

```diff
@@ -649,5 +650,25 @@ describe('class MongoClient', function () {
       expect(endEvents).to.have.lengthOf(1);
       expect(endEvents[0]).to.have.property('reply', undefined); // noReponse: true
     });
+
+    context('when server selection would return no servers', () => {
+      const serverDescription = new ServerDescription('a:1');
+
+      it('short circuits and does not end sessions', async () => {
```

Comment:
This test will hang and wait for server selection without the fix in, thus reproducing the path.

Reply:
Just to demonstrate that `endSession` does not do anything but bookkeeping, let's assert that the `client.s.sessionPool` still have one session in it both before and after close.

Reply:
Added.